### ISSUE TITLE
haskellPackages.clay has missing dependencies

### ISSUE THREAD

**Issue by JonathanReeve** [NixOS/nixpkgs#54958]
## Issue description

When trying to install anything that depends on haskellPackages.clay, the builder fails, complaining that: `Setup: Encountered missing dependencies:
hspec >=2.2.0 && <2.6, hspec-discover >=2.2.0 && <2.6`

### Steps to reproduce

Try running `nix-env -iA nixos.haskellPackages.clay`

## Technical details

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the
results.
 - system: `"x86_64-linux"`
 - host os: `Linux 4.19.16, NixOS, 19.03pre166987.bc41317e243 (Koi)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.2`
 - channels(jon): `"home-manager, nixos-19.03pre166987.bc41317e243"`
 - channels(root): `"nixos-19.03pre166987.bc41317e243"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos/nixpkgs`

**Comment by wizzup** [comment]
If you can take some [risk](https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/haskell-modules/lib.nix#L79) of breaking things, you can do `jailbreak` override

```
# ~/.config/nixpkgs/config.nix

{ pkgs ? import <nixpkgs> {} }:
{
  packageOverrides = super: {

    haskellPackages = {
      clay = super.haskell.lib.doJailbreak super.haskellPackages.clay;
    };
  };
}
 
```
*Disclaimer*: I don't know if this is the proper way to override `haskellPackages`, normally I would instead introduce new namespace like `myHaskell` and use it like `nix-shell -p myHaskell.clay` to make sure I don't pollute the official namespace.

**Comment by peti** [comment]
The problem is that upstream does not support recent versions of these packages. `hspec`, for example, is currently at version "2.6.1". Upstream should publish a new version of `clay` that remedies these issues, then it will compile successfully in Nixpkgs.

**Comment by JonathanReeve** [comment]
This bug still exists, as far I can tell. How can we fix this package/derivation so that it works?

**Comment by andgate** [comment]
There is a PR(sebastiaanvisser/clay#187) on clay to update the version bounds.

**Comment by turion** [comment]
That PR is now closed, and the updated package uploaded. On the next automatic hackage crawl, this should be automatically resolved.

**Comment by turion** [comment]
@peti is it correct that this will be resolved automatically? The new version (0.13.2) is released since some time already.

**Comment by turion** [comment]
It was necessary to manually unmark the package as broken. I had no idea. This will be fixed with https://github.com/NixOS/nixpkgs/pull/75429.
